arraycontext/impl/pyopencl/fake_numpy.py

```diff
@@ -239,32 +240,33 @@ def all(self, a):
         return result
 
     def array_equal(self, a, b):
-        def as_device_scalar(bool_value):
-            import numpy as np
-            return self._array_context.from_numpy(
-                np.array(int(bool_value), dtype=np.int8))
+        actx = self._array_context
+        queue = actx.queue
+
+        # NOTE: pyopencl doesn't like `bool` much, so use `int8` instead
+        false = actx.from_numpy(np.int8(False))
```

Comment:
After inducer/pyopencl#518 and #113, I'm not sure this needs to be `from_numpy`. Thoughts?

Reply:
Sounds good!

I'm not too bothered either way, since, like `norm`, the rule would be "if we don't do any device computation, we return a host thing", which seems reasonable enough.